kibitzr/transformer/html.py

```diff
@@ -83,18 +83,33 @@ def xpath_selector(selector, html, select_all):
     # see https://github.com/kibitzr/kibitzr/issues/47
     encoded = html.encode('utf-8')
     root = dlxml.fromstring(encoded, parser=etree.HTMLParser())
-    elements = root.xpath(selector)
-    if not elements:
+    xpath_results = root.xpath(selector)
+    if not xpath_results:
         logger.warning('XPath selector not found: %r', selector)
         return False, html
-    if select_all is False:
-        elements = elements[0:1]
 
-    elements = [re.sub(r'\s+', ' ',
-                dlxml.tostring(ele, method='html', encoding='unicode')).strip()
-                for ele in elements]
-
-    return True, u"\n".join(six.text_type(x) for x in elements)
+    if isinstance(xpath_results, list):
+        if select_all is False:
+            xpath_results = xpath_results[0:1]
+    else:
+        xpath_results = [xpath_results]
+
+    # Serialize xpath_results
```

Comment:
Please add a comment with link to [lxml xpath docs](https://lxml.de/xpathxslt.html#xpath-return-values)